ch_tools/chadmin/cli/table_group.py

```diff
@@ -242,20 +252,49 @@ def columns_command(ctx, database_name, table_name):
     default=False,
     help="Enable dry run mode and do not perform any modifying actions.",
 )
+@option(
+    "--detached",
+    is_flag=True,
+    help="Delete detached tables (with nonreplicated engine).",
+)
+@constraint(
+    If(IsSet("detached"), then=require_all),
+    [
+        "database_name",
+        "table_name",
+    ],
+)
+@constraint(
+    If(IsSet("detached"), then=accept_none),
+    [
+        "on_cluster",
+    ],
+)
 @pass_context
 def delete_command(
     ctx,
     _all,
     on_cluster,
     sync_mode,
     dry_run,
+    detached,
+    database_name,
+    table_name,
     **kwargs,
 ):
     """
     Delete one or several tables.
     """
     cluster = get_cluster_name(ctx) if on_cluster else None
 
+    if detached:
+        delete_detached_table(
```

Comment:
Suggest to implement support of dry-run mode for deleting detached tables. Or at least mark --dry-run option as incompatible with --detached.

Reply:
I will mark dry as incompatible with --detached.